library/src/main/java/com/schibsted/spain/barista/cleardata/ClearDatabaseRule.kt

```diff
@@ -0,0 +1,48 @@
+package com.schibsted.spain.barista.cleardata
+
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+import java.io.File
+
+/**
+ * This rule clears all app's Databases before running each test
+ */
+class ClearDatabaseRule(private val databaseOperations: DatabaseOperations = DatabaseOperations()) : TestRule {
+
+    companion object {
+        @JvmField
+        internal val UNWANTED_FILENAME_SUFFIXES = arrayOf("-journal", "-shm", "-uid", "-wal")
+    }
+
+    override fun apply(base: Statement, description: Description): Statement {
+        return object : Statement() {
+            override fun evaluate() {
+                clearDatabases()
+                base.evaluate()
+                clearDatabases()
```

Comment:
Why do you need to call twice this method? Looks weird...

Reply:
- Before: Clear the database so our test runs on a clean environment.
- After: Clear the database to leave the environment clean for the next test.

If everyone did it you could just do it at the beginning or at the end. But since tests are isolated and can't trust the other tests, it's good to make sure.